ccl/changefeedccl: TestAvroWithRegionalTable failed

ccl/changefeedccl.TestAvroWithRegionalTable [failed](https://tanzanite.cluster.engflow.com/invocations/default/627737a6-2d5c-4a18-b080-067ff49615c2?testReportRun=2&testReportShard=6&testReportAttempt=1#targets-Ly9wa2cvY2NsL2NoYW5nZWZlZWRjY2w6Y2hhbmdlZmVlZGNjbF90ZXN0) on release-24.1 @ [9961abaaa0fdc6abf036efe1c4cffb71c8bb126e](https://github.com/cockroachdb/cockroach/commits/9961abaaa0fdc6abf036efe1c4cffb71c8bb126e):


```
=== RUN   TestAvroWithRegionalTable/key_only/overrideWithSingleWorker=true
    encoder_test.go:1291: overriding number of parallel workers to one

pkg/ccl/changefeedccl/testfeed_test.go:671: (newDepInjector)
	NOTICE: .DistSQLServer() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().DistSQLServer() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization

pkg/ccl/changefeedccl/testfeed_test.go:671: (newDepInjector)
	NOTICE: .DistSQLServer() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().DistSQLServer() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization

pkg/ccl/changefeedccl/testfeed_test.go:671: (newDepInjector)
	NOTICE: .DistSQLServer() called via implicit interface ApplicationLayerInterface;
HINT: consider using .ApplicationLayer().DistSQLServer() instead.
TIP: consider replacing the test server initialization from:
    ts, ... := serverutils.StartServer(t, ...)
    defer ts.Stopper().Stop(...)
to:
    srv, ... := serverutils.StartServer(t, ...)
    defer srv.Stopper().Stop(...)
    ts := srv.ApplicationLayer()

See also: https://go.crdb.dev/p/testserver-and-cluster-virtualization
    encoder_test.go:1309: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:199
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/helpers_test.go:274
        	            				github.com/cockroachdb/cockroach/pkg/ccl/changefeedccl/encoder_test.go:1309
        	            				github.com/cockroachdb/cockroach/pkg/testutils/subtest.go:28
        	Error:      	Received unexpected error:
        	            	cannot unmarshal schema JSON: unexpected end of JSON input
        	Test:       	TestAvroWithRegionalTable/key_only/overrideWithSingleWorker=true
        --- FAIL: TestAvroWithRegionalTable/key_only/overrideWithSingleWorker=true (30.29s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=6</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestAvroWithRegionalTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39012

--- a/pkg/ccl/changefeedccl/encoder_test.go
+++ b/pkg/ccl/changefeedccl/encoder_test.go
@@ -1285,6 +1285,10 @@ func TestAvroWithRegionalTable(t *testing.T) {
 			// are used. With one worker, the cache is forced to be used during
 			// encoding for the second row.
 			testutils.RunTrueAndFalse(t, "overrideWithSingleWorker", func(t *testing.T, overrideWithSingleWorker bool) {
+				// Clear the singleton cache to avoid pollution from other tests.
+				// This needs to be done here since this test doesn't use the
+				// cdcTest helper function.
+				TestingClearSchemaRegistrySingleton()
 				cluster, db, cleanup := startTestCluster(t)
 				defer cleanup()
 				if overrideWithSingleWorker {

